### ISSUE TITLE
Password is not hidden

### ISSUE THREAD

**Issue by perry-mitchell** [jdxcode/password-prompt#5]
When prompting for a new password on 1.0.3 or 1.0.4, the password is printed:

![image](https://user-images.githubusercontent.com/3869469/36719979-4415a15a-1baf-11e8-9a3a-0f4a0ce006c3.png)

I'm running node 8 on Mac High Sierra in iterm2. I'm using this to reproduce:

```javascript
const prompt = require("password-prompt");
prompt("Password: ", { method: "hide" });
```

**Comment by rocketmonkeys** [comment]
I get a similar problem on windows 7... 

During typing it shows:

     password: abcdefg

And after you hit enter it shows:

     password:*******
     password: abcdefg

Doesn't hide anything.

**Comment by oxfn** [comment]
@rocketmonkeys To get what you expect you have to call `prompt('What: ', { method: 'hide' })`.

This is not obvious, because '**mask**' supposes that text is hidden immediately after keydown or paste (this is how '**hide**' works now) and '**hide**' supposes that we see nothing on input.

@jdxcode, what do you think?

**Comment by jdxcode** [comment]
that's right. The intention is for something like a 2fa key where it's fine to display the key but after it's pasted it can be masked. I suppose there could be yet another display type that masks as you type as well. I'm not sure what it should be named though.

**Comment by oxfn** [comment]
> I suppose there could be yet another display type that masks as you type as well. I'm not sure what it should be named though.

This would be great!
How about **suppress**? I think verb is better for consistency.

Also I would suggest to rename methods in future major release:
 - **hide** -> **mask** (default, display asterisks immediately)
 - **mask** -> **preview** (hide input after enter, name looks clearer to me)
 - **suppress** (do not show anything)

**Comment by jdxcode** [comment]
we can't change them, that would wreak havoc

**Comment by oxfn** [comment]
@perry-mitchell I use approximately the same software as you reported and did not face such a problem:

    macOS High Sierra 10.13
    iTerm2 v3.2.0 + tmux v2.7
    node v10.7.0

Maybe bug is in node version or some low level library

**Comment by zypA13510** [comment]
@jdxcode 
> we can't change them, that would wreak havoc

Not sure I understand. Why can't we change it in a later **major** version? Major versions are meant for breaking changes, right? And in the case of "mask", oxfn's proposal actually makes them less visible, so I don't see any significant security issue. If people upgrade to the next major without reading the changelog or update their code, it's their problem anyway.
